Custom board build action doesn't work

Trying to make a custom build with the custom build action, I get a failure, with the following error:
```
Run make -j4 $FLAGS BOARD="$BOARD" DEBUG=$DEBUG TRANSLATION="$TRANSLATION"
  make -j4 $FLAGS BOARD="$BOARD" DEBUG=$DEBUG TRANSLATION="$TRANSLATION"
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.13.2/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.13.2/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.2/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.2/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.2/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.13.2/x64/lib
    TRANSLATION: en_US
    BOARD: adafruit_feather_esp32s3_reverse_tft
    FLAGS: 
    DEBUG: 0
make: *** No targets specified and no makefile found.  Stop.
Error: Process completed with exit code 2.
```
This seems to be caused by #10026 which removed `board-to-port` from [`deps/ports/action.yml`](https://github.com/adafruit/circuitpython/blob/main/.github/actions/deps/ports/action.yml) requiring instead to input the port, without adding a `port` entry in the custom action. Personally I would prefer NOT to add more complexity to the custom build action form and instead revert to the previous behavior, or to find the port as before if absent.

--- a/.github/workflows/build-board-custom.yml
+++ b/.github/workflows/build-board-custom.yml
@@ -73,11 +73,18 @@ jobs:
       uses: actions/setup-python@v5
       with:
         python-version: 3.x
+    - name: Board to port
+      id: board-to-port
+      run: |
+        PORT=$(python tools/board_to_port.py "${{ inputs.board }}")
+        echo "port=$PORT" >> $GITHUB_OUTPUT
+      shell: bash
     - name: Set up port
       id: set-up-port
       uses: ./.github/actions/deps/ports
       with:
         board: ${{ inputs.board }}
+        port: ${{ steps.board-to-port.outputs.port }}
     - name: Set up submodules
       id: set-up-submodules
       uses: ./.github/actions/deps/submodules
@@ -88,7 +95,7 @@ jobs:
       uses: ./.github/actions/deps/external
       with:
         action: cache
-        port: ${{ steps.set-up-port.outputs.port }}
+        port: ${{ steps.board-to-port.outputs.port }}
     - name: Set up mpy-cross
       if: steps.set-up-submodules.outputs.frozen == 'True'
       uses: ./.github/actions/mpy_cross
@@ -115,9 +122,9 @@ jobs:
         FLAGS: ${{ inputs.flags }}
         DEBUG: ${{ inputs.debug && '1' || '0' }}
       run: make -j4 $FLAGS BOARD="$BOARD" DEBUG=$DEBUG TRANSLATION="$TRANSLATION"
-      working-directory: ports/${{ steps.set-up-port.outputs.port }}
+      working-directory: ports/${{ steps.board-to-port.outputs.port }}
     - name: Upload artifact
       uses: actions/upload-artifact@v4
       with:
         name: ${{ inputs.board }}-${{ inputs.language }}-${{ inputs.version }}${{ inputs.flags != '' && '-custom' || '' }}${{ inputs.debug && '-debug' || '' }}
-        path: ports/${{ steps.set-up-port.outputs.port }}/build-${{ inputs.board }}/firmware.*
+        path: ports/${{ steps.board-to-port.outputs.port }}/build-${{ inputs.board }}/firmware.*

--- a/tools/board_to_port.py
+++ b/tools/board_to_port.py
@@ -1,0 +1,22 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2025 CircuitPython contributors (https://github.com/adafruit/circuitpython/graphs/contributors)
+#
+# SPDX-License-Identifier: MIT
+
+import sys
+from pathlib import Path
+
+docs = Path(__file__).parent.parent / "docs"
+sys.path.append(str(docs))
+from shared_bindings_matrix import get_board_mapping
+
+board = sys.argv[1]
+
+board_mapping = get_board_mapping()
+if board in board_mapping:
+    board_info = board_mapping[board]
+    print(board_info["port"])
+    sys.exit(0)
+
+raise ValueError(f"No port directory associated with the board tag given {board}.")

